src/fm.py

```diff
@@ -33,5 +33,172 @@ def main():
         print(f"Search {args.genome} for {args.reads}")
 
 
+def lower(a: str, x: str, sa: memoryview, lo: int, hi: int, offset: int) -> int:
+    """Finds the lower bound of `a` at `offset` in the block defined by `lo:hi`."""
+    while lo < hi:  # Search in sa[lo:hi]
+        m = (lo + hi) // 2
+        if x[sa[m] + offset % len(x)] < a: # compare at column offset in sa
```

Comment:
Let's imagine `sa[m] == 4`, `len(x) == 6` and `offset == 3`. Then `offset % len(x) == 3` and `sa[m] + offset % len(x) == 7`. If `len(x) == 6`, `x[7]` usually will not end well.

Reply:
What I'm saying is `x[(sa[m]+offset) % len(x)]` is a better choice.